### PR TITLE
print_mounted_endpoints

### LINKED ISSUE #1453: Print routes when mounting them
This feature involves a few changes:

1. Enhance the generated `Server` struct in `gen/http/<service>/server/server.go` with a new field that captures the mounts.
2. Enhance the generated server `main.go` so  that before starting to listen to requests it iterates through the server mounted handlers and logs them.

Item 1 could look something like:
```go
type (
    Server struct {
        // Service is the name of the service served by this HTTP server.
       Service string
        // Mounts lists the mounted HTTP handlers.
        Mounts []*Mount
        .....  // HTTP handlers as currently generated
    }

    Mount struct {
        // Method is the name of the service method served by the mounted HTTP handler.
        Method string
        // Verb is the HTTP method used to match requests to the mounted handler.
        Verb string
        // Pattern is the HTTP request path pattern used to match requests to the mounted handler.
        Pattern string
        // Handler is the mounted HTTP handler.
        Handler http.Handler
    }
)
```
and 2 could look like:
```go
    // Start HTTP server using default configuration, change the code to
    // configure the server as required by your service.
    srv := &http.Server{Addr: *addr, Handler: handler}
    go func() {
        for _, m := range <service>Server.Mounts {
            logger.Printf("[INFO] service %q method %q mounted on %s %s", 
                <service>Server.Service, m.Method, m.Verb, m.Pattern)
        }
        logger.Printf("[INFO] listening on %s", *addr)
        errc <- srv.ListenAndServe()
    }()
}()
```
Also note that the current code generator optimizes the case where there is no need for a `Server` struct to be generated if there are only file servers exposed by the service and no regular method. This optimization needs to be removed and the file server case handled properly as well.

### PR DESCRIPTION
**Pull request by nitinmohan87**
Fixes #1453

### CODE REVIEW

File: design/api.go [hunk 1]
```diff
@@ -115,6 +115,9 @@ func (a *APIExpr) Random() *Random {
 // EvalName is the qualified name of the expression.
 func (a *APIExpr) EvalName() string { return "API " + a.Name }
 
+// Hash returns a unique hash value for a.
+func (a *APIExpr) Hash() string { return a.Name }
```
Comment (raphael) [inline design/api.go]: We may need to specify a prefix here so there's no risk of clashing with other constructs.

File: examples/cellar/gen/http/swagger/server/server.go [hunk 2]
```diff
@@ -9,11 +9,46 @@
...
+) *Server {
+	return &Server{
+		Mounts: []*MountPoint{
+			{"../../gen/http/openapi.json", "GET", "/swagger/swagger.json"},
+		},
+	}
+}
+
+// Service returns the name of the service served.
+func (s *Server) Service() string { return "swagger" }
```
Comment (raphael) [inline examples/cellar/gen/http/swagger/server/server.go]: Do we need this function?

Comment (nitinmohan87) [reply examples/cellar/gen/http/swagger/server/server.go]: Yes.. It is used here - https://github.com/goadesign/goa/blob/bc1a832c804fb1ebca28ab8f4289b213c1ce3f00/examples/cellar/cmd/cellarsvc/main.go#L154

**Comment by raphael** [comment]
Looks great, thank you!

**Pull request merged**